### PR TITLE
Simplify configuration logic

### PR DESCRIPTION
**Pull request by kytrinyx**
I pulled some of the things I like from the spike out into separate commits, and applied them here.

All of it boils down to simplifying the configuration logic:

- remove the unmarshaling from client.Do(), since we want more control over the error handling and error messages
- get rid of the API config, folding in the base url into the user config
- pass token and base url directly into the API client, as they don't actually need any other configuration

The bottom line is that now the _only_ package that imports the `config` package is the `cmd` package, which will simplify a few things going forward.

FYI @nywilken I'm going to go ahead and merge this so that I can work on top of it.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/spf13/viper"
 )
 
+var (
+	defaultBaseURL = "https://v2.exercism.io/api/v1"
```
Comment (nywilken) [inline config/config.go]: I find this to be a bit confusing seeing as config/config is a generic wrapper around config with no specific exercism data. In my opinion it is probably better off in `user_config.go`

File: cmd/configure_test.go [hunk 2]
```diff
@@ -32,11 +32,11 @@ func TestConfigure(t *testing.T) {
 		testCase{
 			desc:           "It overwrites the flags that are passed, without losing the ones that are not.",
 			args:           []string{"fakeapp", "configure", "--token", "c", "--skip-auth"},
-			existingUsrCfg: &config.UserConfig{Token: "token-c", Workspace: "/workspace-c"},
-			expectedUsrCfg: &config.UserConfig{Token: "c", Workspace: "/workspace-c"},
+			existingUsrCfg: &config.UserConfig{Token: "token-c", Workspace: "/workspace-c", APIBaseURL: "http://example.com"},
```
Comment (nywilken) [inline cmd/configure_test.go]: For a second I thought the APIBaseURL was some how being obtained from a previous testcase using `http://example.com`. Maybe having a workspace and APIBaseURL that is different from other test cases or something like `Workspace: "/unmodified-workspace-path", APIBaseURL: "http://unmodified.example.com/"` would help illustrate the point. Or a description like "Unspecified flags do not overwrite previously configured settings"

**Comment by kytrinyx** [comment]
> The PR description seems like the changes should be doing more, but guessing you might still be working on it.

Yes! I just wanted to get in the basics before it becomes massively difficult to review.

File: config/config.go (hunk 1 above)
Comment (kytrinyx) [reply config/config.go]: Yepp, that's a good point. I'll make that change.

File: cmd/configure_test.go (hunk 2 above)
Comment (kytrinyx) [reply cmd/configure_test.go]: Nice, yeah, that would be much clearer.

Comment (kytrinyx) [reply cmd/configure_test.go]: I went ahead and did a complete overhaul of the test data values in that test to help clarify intention throughout. I like it a lot better! Thanks for the suggestion.

Comment (nywilken) [reply cmd/configure_test.go]: Looks good!
